libs/wingsdk/src/target-sim/dynamodb-table.inflight.ts

```diff
@@ -0,0 +1,117 @@
+import { CreateTableCommand, DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { v4 as uuidv4 } from "uuid";
+import {
+  DynamodbTableAttributes,
+  DynamodbTableSchema,
+} from "./schema-resources";
+import { DynamodbTableClientBase } from "../ex";
+import { runCommand, runDockerImage } from "../shared/misc";
+import {
+  ISimulatorContext,
+  ISimulatorResourceInstance,
+} from "../testing/simulator";
+
+const MAX_CREATE_TABLE_COMMAND_ATTEMPTS = 20;
+
+export class DynamodbTable
+  extends DynamodbTableClientBase
+  implements ISimulatorResourceInstance
+{
+  private containerName: string;
+  private readonly WING_DYNAMODB_IMAGE =
+    process.env.WING_DYNAMODB_IMAGE ?? "amazon/dynamodb-local:2.0.0";
+  private readonly context: ISimulatorContext;
+  private client?: DynamoDBClient;
+
+  public constructor(
+    props: DynamodbTableSchema["props"],
+    context: ISimulatorContext
+  ) {
+    super(props.name, props.primaryKey);
+
+    this.context = context;
+    this.containerName = `wing-sim-dynamodb-${this.context.resourcePath.replace(
+      /\//g,
+      "."
+    )}-${uuidv4()}`;
+  }
+
+  public async init(): Promise<DynamodbTableAttributes> {
+    try {
+      const { hostPort } = await runDockerImage({
+        imageName: this.WING_DYNAMODB_IMAGE,
+        containerName: this.containerName,
+        containerPort: "8000",
+      });
+
+      // dynamodb url based on host port
+      this.client = new DynamoDBClient({
+        region: "local",
+        credentials: {
+          accessKeyId: "x",
+          secretAccessKey: "y",
+        },
+        endpoint: `http://0.0.0.0:${hostPort}`,
+      });
+
+      await this.createTable();
+
+      return {};
+    } catch (e) {
+      throw Error(`Error setting up Dynamodb resource simulation (${e})
+      - Make sure you have docker installed and running`);
+    }
+  }
+
+  public async cleanup(): Promise<void> {
+    // disconnect from the dynamodb server
+    this.client?.destroy();
+    // stop the dynamodb container
+    await runCommand("docker", ["rm", "-f", `${this.containerName}`]);
+  }
```

Comment:
I wonder if the cleanup gets called during kill signals. @Chriscbr ?

If it doesn't, we may want to use something like the `death` package:

```ts
      const { default: DEATH } = await import("death");
      DEATH(() => {
        // Do all cleanups
      });
```